src/core/algorithms/gfd/gfd.h

```diff
@@ -43,5 +53,11 @@ class Gfd {
         conclusion_ = conclusion;
     }

```

Comment:
Here we need to dive deeper into how those setters are used, because we have three options:
1. Pass by value and do `pattern_ = std::move(pattern);`, which will:
- with lvalues: one copy and one move
- with rvalues: two moves

2. Pass by const reference and do `pattern_ = std::move(pattern);`, which will:
- always be a one copy (won't work with rvalues)

3. Pass by universal reference and do `pattern_ = std::forward<T>(pattern);` 
- with lvalues: one copy
- with rvalues: one move


We use that code like so:
```cpp
Gfd ReadGfd(std::istream& stream) {
    std::vector<Literal> premises = ParseLiterals(stream);
    std::vector<Literal> conclusion = ParseLiterals(stream);
    graph_t pattern = ReadGraph(stream);
    Gfd result = Gfd();
    result.SetPattern(pattern);
    result.SetPremises(premises);
    result.SetConclusion(conclusion);
    return result;
};
```
So we have a function `ParseLiterals` that returns `std::vector<Literal>`, but instead of just passing it to a setter, we'll **copy it first** to a local variable, then **copy it second** time in the setter.

By utilizing universal references with `std::forward` we can completely avoid **any copies** by just passing rvalue directly:
```cpp
Gfd ReadGfd(std::istream& stream) {
    Gfd result{};
    result.SetPremises(ParseLiterals(stream));
    result.SetConclusion(ParseLiterals(stream));
    result.SetPattern(ReadGraph(stream));
    return result;
};
```